charms/jupyter-ui/src/charm.py

```diff
@@ -242,6 +251,42 @@ def _deploy_k8s_resources(self) -> None:
             raise ErrorWithStatus("K8S resources creation failed", BlockedStatus)
         self.model.unit.status = MaintenanceStatus("K8S resources created")
 
+    def _get_images_config(self, key: str) -> OptionsWithDefault:
+        """Load, validate, render and return the config value for Notebook Images.
+
+        The config options for the notebook images will have an empty value by default.
+        Thus the default value will be read from the corresponding files with the default
+        values.
+        """
+        default_images_file = ""
+        if key == JUPYTER_IMAGES_CONFIG:
+            default_images_file = DEFAULT_JUPYTER_IMAGES_FILE
+        if key == RSTUDIO_IMAGES_CONFIG:
+            default_images_file = DEFAULT_RSTUDIO_IMAGES_FILE
+        if key == VSCODE_IMAGES_CONFIG:
+            default_images_file = DEFAULT_VSCODE_IMAGES_FILE
+
+        try:
+            images = yaml.safe_load(str(self.model.config.get(key, "")))
+            self.logger.info(f"Model config value: {self.model.config.get(key, '')}")
+        except ParserError as e:
+            raise ConfigValidationError(f"Config value is not a yaml for {key}.") from e
```

Comment:
issue(blocking): this exception is not getting caught own the charm execution flow (in the `main` handler), it will end up being an unhandled exception and cause the charm to error out. It used to be handled in `_get_list_config` by ignoring the incorrect config and logging a warning. We could either replicate that behavior, or go to `BlockedStatus` asking the user to change the config. I'm leaning more towards the previous behavior to keep consistent with the other configs.